book/listings/pharmacokinetic/adal01.qmd

```diff
@@ -0,0 +1,120 @@
+---
+title: ADAL01
+subtitle: Listing of Anti-Drug Antibody Data for Patients with At Least One ADA Sample Datum by Treatment
+---
+
+------------------------------------------------------------------------
+
+{{< include ../../test-utils/envir_hook.qmd >}}
+
+::: panel-tabset
+## Data Setup
+
+```{r setup, message=FALSE}
+#| code-fold: show
+
+library(dplyr)
+library(rlistings)
+library(scda)
+
+adpc <- synthetic_cdisc_dataset("latest", "adpc")
+adab <- synthetic_cdisc_dataset("latest", "adab")
+
+trt <- "A: Drug X"
+min_titer_ada <- 1.10
+min_titer_nab <- 0
+min_conc <- 0
+
+adpc_f <- adpc %>% filter(PARAM == "Plasma Drug X")
+
+if (unique(adab$RELTMU) == "day") adab$NFRLT <- adab$NFRLT * 24
+
+drug_a <- unique(adab$PARCAT1)[1]
+drugcd <- unique(adab$PARAMCD[adab$PARAM == "Antibody titer units"])[1]
+conc_u <- unique(adab$AVALU[adab$PARAM == "Antibody titer units"])[1]
+
+adpc_f <- adpc_f %>% select(USUBJID, NFRLT, AVAL)
+adab_f <- adab %>%
+  filter(ARM == trt) %>%
+  select(-PARAMCD, -PARCAT1, -AVALC, -AVALU) %>%
+  left_join(
+    adpc_f,
+    by = c("USUBJID", "NFRLT"),
+    suffix = c("_ab", "_pk")
+  ) %>%
+  filter(!is.na(AVAL_ab))
+
+adab_f <- adab_f %>%
+  tidyr::pivot_wider(
+    names_from = PARAM,
+    values_from = AVAL_ab
+  ) %>%
+  rename(
+    RES_ADA = `ADA interpreted per sample result`,
+    TU_ADA = `Antibody titer units`,
+    RES_NAB = `NAB interpreted per sample result`,
+    TU_NAB = `Neutralizing Antibody titer units`
+  )
+
+out <- adab_f %>%
+  mutate_at(
+    c("NFRLT", "TU_ADA", "TU_NAB", "AVAL_pk"),
+    ~ ifelse(is.na(.), replace(., is.na(.), "N/A"), format(round(., 2), nsmall = 2))
+  ) %>%
+  mutate(NFRLT = as.numeric(NFRLT)) %>%
+  mutate(
+    RES_ADA = ifelse(RES_ADA == 1, "Positive", "Negative"),
+    RES_NAB = ifelse(RES_NAB == 1, "Positive", "Negative"),
+    TU_ADA = ifelse(TU_ADA <= min_titer_ada, "BLQ", TU_ADA),
+    TU_NAB = ifelse(TU_NAB <= min_titer_nab, "BLQ", TU_NAB),
+    AVAL_pk = ifelse(AVAL_pk <= min_conc, "BLQ", AVAL_pk)
+  ) %>%
+  select(
+    USUBJID, VISIT, ISTPT, NFRLT, RES_ADA, TU_ADA, RES_NAB, TU_NAB,
+    AVAL_pk
+  )
+
+var_labels(out) <- names(out)
+
+out <- out %>%
+  var_relabel(
+    USUBJID = "Subject ID",
+    VISIT = "Visit",
+    ISTPT = "Timepoint",
+    NFRLT = "Nominal\nTime\n(hr)",
+    RES_ADA = "Sample\nADA\nResult",
+    TU_ADA = "ADA\nTiter\nUnits\n(1)",
+    RES_NAB = "Sample\nNeutralizing\nAntibody\n(NAb) Result",
+    TU_NAB = "NAb\nTiter\nUnits\n(2)",
+    AVAL_pk = paste0("Drug\nConcentration\n(", conc_u, ") (3)")
+  ) %>%
```

Comment:
@edelarua, patient treatment emergent ADA status PARAMCD = 'ADASTAT1' and Patent Treatment Emergent NAB status PARAMCD = 'ADASTAT2' are missing. I may confused you last time. We had all the columns below except the column "Charcterization Results".
![Screen Shot 2023-09-20 at 2 50 42 PM](https://github.com/insightsengineering/tlg-catalog/assets/24441440/5d0c4320-af46-4eac-9843-6322d3c99603)

Reply:
@qiliu1013 in that case [my earlier comment](https://github.com/insightsengineering/tlg-catalog/pull/116#issuecomment-1712110339) still applies - values for these two columns are not generated within `random.cdisc.data` so these datasets will need to be updated if we want to include these two columns in the table.